MultiOracle.md

```diff
@@ -0,0 +1,450 @@
+only the primary oracle (as usual) and where `xCET` and `middleCET` are aggregations of secondary oracles such that
+
+* All secondary oracles are included in exactly one of `xCET` or `middleCET` and
+* `xCET` is non-empty.
+
+To summarize, for n-of-n oracle support, the 2-of-2 algorithm is run where the primary oracle's
+
+* Small Middle CETs result in a single aggregate CET
+* Small non-Middle CETs result in `2^(n-1)` aggregate CETs
+* Large CETs result in a total of `2^n - 1` aggregate CETs
+  * One aggregate middle CET
+  * `2^(n-1) - 1` aggregate non-middle CETs for each side
+    * Totaling `2^n - 2` non-middle aggregate CETs.
+
+#### Analysis
+
+Luckily, there are very few large CETs in any given DLC.
+Thus, the primary multiplier to be concerned with is that of small non-middle CETs.
+
+If the difference between `maxErrorExp` and `minFailExp` is only one, then all small
+CETs will be non-middle so that additional oracles are guaranteed to lead to exponential
+growth in this multiplier (though one should note that exponential growth is already likely
+when using the below t-of-n scheme so that this may not be a primary concern).
+
+If, on the other hand, the difference between `maxErrorExp` and `minFailExp` is larger
+than one, then the majority of CETs are expected to be middle CETs with exponentially larger
+numbers of middle CETs the bigger the difference between those two parameters.
+
+Therefore, there is a trade-off between the tightness of error bounds (i.e. the size of the non-
+guaranteed successes or failures between `minFail` and `maxError`)  and the number of CETs
+when using more than two oracles.
+
+It should also be noted that the actual values of `minFailExp` and `maxErrorExp` are not part of
+this trade-off, only their relative difference.
+For example, having `minFail=32` and `maxError = 128` should yield similar results to having
+`minFail = 256` and `maxError = 1024`.
+
+### t-of-n Oracles with Bounded Error
+
+Just as in the [t-of-n Oracles for enumerated outcomes](#t-of-n-oracles) case, to support a threshold t-of-n oracles for
+numeric outcomes with allowed (bounded) error, we simply run the t-of-t algorithm above on all of the
+`n C t` possible combinations of `t` oracles.
+
+The only additional work that must be done is that there must be a decision procedure to determine
+which oracle in a given group of `t` oracles is primary, and additionally what values should be used
+for `minFail` and `maxError` for a given group of `t` oracles (as this can vary between groups).
+
+The simplest candidate would be a negotiated ordered ranking of the oracles where the highest ranking
+oracle in any group of `t` oracles is chosen as primary.
+The values `minFail` and `maxError` can also be computed as some function of the rankings of a group
+of `t` oracles, or else they can be set as constant parameters across all groupings.
+
+TODO: Decide how this is done and communicated. 
```

Comment:
Isn't this done, just ranked by the order of them in the oracle info

Reply:
That is how we currently do it in bitcoin-s but I'm open to the idea of doing something more general such as allowing the parameters to vary by choice of `t` oracles as mentioned in the previous paragraph

Reply:
I suppose order-ranked oracle info could be added as a `negotiation_field` to `dlc_accept` message and the `initiator` could choose to either sign or not.